### ISSUE TITLE
Can't access localhost from extension. V 1.52 works fine.

### ISSUE THREAD

**Issue by bobolinks** [microsoft/vscode#123757]
- VS Code Version:

Version: 1.56.1 (Universal)
Commit: e713fe9b05fc24facbec8f34fb1017133858842b
Date: 2021-05-06T10:08:24.325Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Darwin x64 20.3.0

- OS Version:
MacOs Big Sur 11.2.3

Steps to Reproduce:
1.  Checks out this example https://github.com/microsoft/vscode-extension-samples/tree/main/webview-sample
2. Inserts a HTML line like '<script nonce="${nonce}" src="http://localhost:50565/xxx.js"></script>'. 
`<script nonce="${nonce}" src="http://localhost:50565/xxx.js"></script>`
3. Lanuch extension 'webview-sample' and open 'Webview developer tool'.

BTW，
'http://localhost:50565/xxx.js' is accessible from browser, and if i use an external link 'https://github.githubassets.com/assets/optimizely-2a30d723.js' to replace 'http://localhost:50565/xxx.js', it works fine.
And when i debug into service-worker.js, i found my request is passed to processLocalhostRequest, but async callback function 'i' is never called. 

Does this issue occur when all extensions are disabled?: No

**Comment by mjbvz** [comment]
I'm not able to reproduce this. Are you connected to a remote workspace when this happens? Do you see any logs in the developer tools?

**Comment by bobolinks** [comment]
logs:
> 
workbench.desktop.main.js:62 [Extension Host] (node:14672) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.(Use `Code Helper (Renderer) --trace-deprecation ...` to show where the warning was created) (at writeOut (internal/process/warning.js:43:3))
S @ workbench.desktop.main.js:62
The FetchEvent for "https://localhost:50565/xxx.js" resulted in a network error response: the promise was rejected.
xxx.js:1 Failed to load resource: net::ERR_FAILED
service-worker.js?id…ined&serviceWork…:1 Uncaught (in promise) TypeError: Failed to fetch
vscodeexperiments.azureedge.net/experiments/vscode-experiments.json:1 Failed to load resource: net::ERR_TIMED_OUT
DevTools failed to load SourceMap: Could not load content for https://ticino.blob.core.windows.net/sourcemaps/054a9295330880ed74ceaedda236253b4f39a335/core/vs/workbench/workbench.desktop.main.js.map: Load canceled due to load timeout
The FetchEvent for "https://localhost:50565/xxx.js" resulted in a network error response: the promise was rejected.
Promise.then (async)
(anonymous) @ service-worker.js:189
service-worker.js?id….calico-colors&p…:1 Uncaught (in promise) TypeError: Failed to fetch
main.js:116 GET https://localhost:50565/xxx.js net::ERR_FAILED

> I'm not able to reproduce this. Are you connected to a remote workspace when this happens? Do you see any logs in the developer tools?

**Comment by mjbvz** [comment]
Are you connected to a remote workspace when this happens?

**Comment by bobolinks** [comment]
No, it's a local workspace. I guess, may be, it is related to the CSP of new version. Is it changed in version 1.56?



At 2021-05-20 08:39:24, "Matt Bierner" ***@***.***> wrote:

Are you connected to a remote workspace when this happens?
